src/d3a/models/area/event_dispatcher.py

```diff
@@ -162,9 +162,6 @@ def create_area_agents(self, is_spot_market, market):
                 min_offer_age=ConstSettings.IAASettings.MIN_OFFER_AGE
             )
 
-            self._delete_past_agents(self._inter_area_agents)
```

Comment:
So this is not needed any more and was just a precaution?
I see that you moved the deletion of the dispatchers into the area. But who deleted the `_inter_area_agents` ?

Reply:
The deletion of the dispatcher includes the deletion of IAAs too. It is just moving this code to the area class that changed here. 
Reason for this change is the Markets class change in the rotate_markets method. The area_agents were popped there without being deleted. By moving the delete_past_agents to the Area class, I wanted to enforce the agents to be destroyed before the markets, in order to not fall into this pitfall again.

Reply:
Thanks for the explanation!